muspinsim_input/muspinsim_input.xml

```diff
@@ -0,0 +1,423 @@
+                        <param name="min_bound" type="text" value="-inf" label="minimum bound"
+                               help="Optional, default is -inf"/>
+                        <param name="max_bound" type="text" value="inf" label="maximum bound"
+                               help="Optional, default is inf"/>
+                    </repeat>
+                    <param name="fitting_tolerance" type="float" optional='true' value="" label="Tolerance for the fitting."
+                           help="Used as the tol parameter in Scipy's scipy.optimize.minimize method. Does not accept expressions/functions/constants"
+                    />
+                </when>
+                <when value=""/>
+            </conditional>
+        </section>
+    </inputs>
+    <outputs>
+        <data format="txt" name="out_file" from_work_dir="outfile.in" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="out_file_prefix" value="test_1"/>
+            <param name="spin_preset" value="custom"/>
+            <param name="spin" value="H"/>
+            <param name="spin_preset" value="mu"/>
+            <param name="interaction" value="zeeman"/>
+            <param name="zeeman_index" value="1" />
+            <param name="zeeman_vector" value="1 0 0" />
+            <param name="zeeman_index" value="2" />
+            <param name="zeeman_vector" value="2 0 0" />
+            <output name="out_file" file="test_1.in" ftype="txt" compare="diff" />
+        </test>
+        <test>
+            <param name="out_file_prefix" value="test_2"/>
+            <param name="spin_preset" value="e"/>
+            <param name="spin_preset" value="mu"/>
+            <param name="interaction" value="hyperfine"/>
+            <param name="hfine_index" value="1" />
+            <param name="hfine_matrix" value="[1 0 0 sin(10) (5*2) 0 10*pi 5 cos(20)]"/>
+            <param name="time" value="range(0, 0.1)" />
+            <param name="y_axis" value="asymmetry" />
+            <param name="field" value="1.0" />
+            <param name="temperature" value="1.0" />
+            <output name="out_file" file="test_2.in" ftype="txt" compare="diff" />
+        </test>
+        <test>
+            <param name="out_file_prefix" value="test_3"/>
+            <param name="spin_preset" value="custom"/>
+            <param name="spin" value="H"/>
+            <param name="spin_preset" value="mu"/>
+            <param name="spin_preset" value="e"/>
+            <param name="interaction" value="hyperfine"/>
+            <param name="hfine_index" value="2" />
+            <param name="hfine_matrix" value="[580 5  10 5   580 9 10  9   580]"/>
+            <param name="interaction" value="hyperfine"/>
+            <param name="hfine_index" value="3" />
+            <param name="hfine_matrix" value="[(300/2)  3   4*10 ] [3  15*10 6-3+2] [4  5   15 ]"/>
```

Comment:
We could definitely do with a test for an input along the lines of `[[1, 2, 3], [3, 4, 5], [4, 5, 6]]` (i.e. extra brackets and comma separation + a pretty standard way to input matrices)